lib/charms/vault_k8s/v0/vault_autounseal.py

```diff
@@ -0,0 +1,457 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the vault-autounseal relation.
```

Comment:
Could you please provide an example on how the requirer and the provider charms would implement this library?

Reply:
I'm more than happy to do this, but maybe this documentation is already more than we need. I kind of lost sight of this while addressing other comments as well.

Since this library is only something we will use internally (or, at least that's the intention for now), it is more akin to `vault_client` and `vault_tls` which have very minimal documentation.

What do you think?

Reply:
I find these examples useful to understand the design of the lib as I see an example of how it is being used.
But I can get the same information from unit tests, so no need to add it here too.